backend/s3/s3.go

```diff
@@ -3228,9 +3233,15 @@ func s3Connection(ctx context.Context, opt *Options, client *http.Client) (s3Cli
 
 	if opt.V2Auth || opt.Region == "other-v2-signature" {
 		fs.Debugf(nil, "Using v2 auth")
-		options = append(options, func(s3Opt *s3.Options) {
-			s3Opt.HTTPSignerV4 = &v2Signer{opt: opt}
-		})
+		if opt.Provider == "IBMCOS" {
+			options = append(options, func(s3Opt *s3.Options) {
```

Comment:
Probably only do this if `opt.IBMCOSAPIKey` and `opt.IBMCOSInstanceID`  are set?

Reply:
addressed